### DISCUSSION TITLE
Add plugins to docker images?

### DISCUSSION THREAD

**Discussion by nniehoff** [nautobot/nautobot#936] [Ideas]
Today with the Nautobot docker image in order to add plugins you need to build a custom docker container.  I am curious what people think of adding the ability to add plugins dynamically to the container, perhaps with an env var something like:

```no-highlight
NAUTOBOT_PLUGINS="nautobot-chatops==1.5.0,nautobot-capacity-metrics==1.1.0"
```

My hesitation with adding this feature really comes down to security.  Today installing plugins requires a `pip install` of a python package.  Enabling an install during runtime especially in a production environment doesn't sit well with me.  On the other hand, it would be incredibly useful to have an easy way to install plugins without a custom image.

Should we enable this for production or maybe just dev environments and we require another variable `NAUTOBOT_DISCLAIMER="yes I know what I am doing but do it anyway"` type thing to enable this feature.  Again concerned a bit about security here if someone were able to set this variable in a prod environment we would basically be enabling installing any package from pypi.

**Comment by jvanderaa** [comment]
The bigger thing that I believe we need to figure out if doing so is how to modify the `PLUGINS_CONFIG` setting. This is a dictionary and there are plugins that will need to have the settings updated. Such as ChatOps specifically, and others as well.

**Reply by nniehoff** [reply]
I know we want to work more towards 12factor, perhaps now that the [django-environ](https://github.com/joke2k/django-environ) project is getting some love we could revisit that which would provide 1 way to provide a `PLUGINS_CONFIG` value.  Today you can override the entire nautobot_config.py which isn't ideal but it is doable.
